core/primitives/cid/comm_cid.cpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "primitives/cid/comm_cid.hpp"
+#include <libp2p/multi/uvarint.hpp>
+#include "primitives/cid/comm_cid_errors.hpp"
+
+namespace fc::common {
+
+  outcome::result<Multihash> rawMultiHash(FilecoinMultihashCode code,
+                                          gsl::span<const uint8_t> commitment) {
+    UVarint u_int(code);
+    UVarint u_int_len(commitment.size());
+    auto bytes = u_int.toVector();
+    bytes.insert(bytes.end(), u_int_len.toVector().begin(), u_int_len.toVector().end());
+    bytes.insert(bytes.end(), commitment.begin(), commitment.end());
+
+    return Multihash::createFromBytes(bytes);
+  }
+
+  outcome::result<std::pair<FilecoinMultihashCode, gsl::span<const uint8_t>>>
+  readMultiHashFromBuf(gsl::span<const uint8_t> bytes) {
```

Comment:
Why not [libp2p `Multihash::createFromBytes`](https://github.com/soramitsu/libp2p/blob/master/src/multi/multihash.cpp#L65)?
Why not update libp2p impl to use uvarint size, if need it at all?

Reply:
Fixed